src/systems/follow_actor/FollowActor.cc

```diff
@@ -0,0 +1,277 @@
+           << std::endl;
+    return;
+  }
+
+  if (_sdf->HasElement("velocity"))
+    this->dataPtr->velocity = _sdf->Get<double>("velocity");
+
+  if (_sdf->HasElement("min_distance"))
+    this->dataPtr->minDistance = _sdf->Get<double>("min_distance");
+
+  if (_sdf->HasElement("max_distance"))
+    this->dataPtr->maxDistance = _sdf->Get<double>("max_distance");
+
+  if (_sdf->HasElement("animation_x_vel"))
+    this->dataPtr->animationXVel = _sdf->Get<double>("animation_x_vel");
+
+  std::string animationName;
+
+  // If animation not provided, use first one from SDF
+  if (!_sdf->HasElement("animation"))
+  {
+    if (actorComp->Data().AnimationCount() < 1)
+    {
+      ignerr << "Actor SDF doesn't have any animations." << std::endl;
+      return;
+    }
+
+    animationName = actorComp->Data().AnimationByIndex(0)->Name();
+  }
+  else
+  {
+    animationName = _sdf->Get<std::string>("animation");
+  }
+
+  if (animationName.empty())
+  {
+    ignerr << "Can't find actor's animation name." << std::endl;
+    return;
+  }
+
+  auto animationNameComp = _ecm.Component<components::AnimationName>(_entity);
+  if (nullptr == animationNameComp)
+  {
+    _ecm.CreateComponent(_entity, components::AnimationName(animationName));
+  }
+  else
+  {
+    *animationNameComp = components::AnimationName(animationName);
+  }
+  _ecm.SetChanged(_entity,
+      components::AnimationName::typeId, ComponentState::OneTimeChange);
+
+  // Set custom animation time from this plugin
+  auto animTimeComp = _ecm.Component<components::AnimationTime>(_entity);
+  if (nullptr == animTimeComp)
+  {
+    _ecm.CreateComponent(_entity, components::AnimationTime());
+  }
+
+  math::Pose3d initialPose;
+  auto poseComp = _ecm.Component<components::Pose>(_entity);
+  if (nullptr == poseComp)
+  {
+    _ecm.CreateComponent(_entity, components::Pose(
+        math::Pose3d::Zero));
+  }
+  else
+  {
+    initialPose = poseComp->Data();
+
+    // We'll be setting the actor's X/Y pose with respect to the world. So we
+    // zero the current values.
+    auto newPose = initialPose;
+    newPose.Pos().X(0);
+    newPose.Pos().Y(0);
+    *poseComp = components::Pose(newPose);
+  }
+
+  // Having a trajectory pose prevents the actor from moving with the
+  // SDF script
+  auto trajPoseComp = _ecm.Component<components::TrajectoryPose>(_entity);
+  if (nullptr == trajPoseComp)
+  {
+    // Leave Z to the pose component, control only 2D with Trajectory
+    initialPose.Pos().Z(0);
+    _ecm.CreateComponent(_entity, components::TrajectoryPose(initialPose));
+  }
+}
+
+//////////////////////////////////////////////////
+void FollowActor::PreUpdate(const UpdateInfo &_info,
+    EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("FollowActor::PreUpdate");
+
+  if (_info.paused)
+    return;
+
+  // TODO(louise) Throttle this system
+
+  // Time delta
+  std::chrono::duration<double> dtDuration = _info.simTime -
+      this->dataPtr->lastUpdate;
+  double dt = dtDuration.count();
```

Comment:
More a curiosity here, can we just use _info.dt or is there any benefit in calculating the dt manually?

Reply:
Yeah good point. I was planning to throttle this system so it doesn't run at 1000 / 500 Hz. I'll go ahead and do that.